projects/gateway/pkg/services/k8sadmisssion/validating_admission_webhook.go

```
@@ -241,28 +247,21 @@ func (wh *gatewayValidationWebhook) makeAdmissionResponse(ctx context.Context, r
 
 	proxyReports, validationErr := wh.validate(ctx, gvk, ref, req.Object, isDelete)
 
-	success := &v1beta1.AdmissionResponse{
-		Allowed: true,
-	}
+	isUnmarshalErr := validationErr != nil && errors.Is(validationErr, UnmarshalErr)
 
-	if validationErr == nil {
+	// even if validation is set to always accept, we want to fail on unmarshal errors
+	if !isUnmarshalErr && (validationErr == nil || wh.alwaysAccept) {
 		logger.Debug("Succeeded")
```

Can we add more info here, always logging`validationErr` when it is non-nil?

---

we already do that here https://github.com/solo-io/gloo/blob/master/projects/gateway/pkg/services/k8sadmisssion/validating_admission_webhook.go#L258

---

But if it is not an unmarshalling error and `alwaysAccept` is `true`, then we just accept the resource and swallow the error.